From patchwork Tue Feb 16 07:21:08 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Andre Heider <a.heider@gmail.com>
X-Patchwork-Id: 1440839
X-Patchwork-Delegate: ynezz@true.cz
Return-Path: 
 <openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=pass (sender SPF authorized) smtp.mailfrom=lists.openwrt.org
 (client-ip=2001:8b0:10b:1231::1; helo=merlin.infradead.org;
 envelope-from=openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org;
	dkim=pass (2048-bit key;
 secure) header.d=lists.infradead.org header.i=@lists.infradead.org
 header.a=rsa-sha256 header.s=merlin.20170209 header.b=ispz/mik;
	dkim=fail reason="signature verification failed" (2048-bit key;
 unprotected) header.d=gmail.com header.i=@gmail.com header.a=rsa-sha256
 header.s=20161025 header.b=oUlg029B;
	dkim-atps=neutral
Received: from merlin.infradead.org (merlin.infradead.org
 [IPv6:2001:8b0:10b:1231::1])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest
 SHA256)
	(No client certificate requested)
	by ozlabs.org (Postfix) with ESMTPS id 4DfsrZ2Spmz9sVF
	for <incoming@patchwork.ozlabs.org>; Tue, 16 Feb 2021 18:23:10 +1100 (AEDT)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=merlin.20170209; h=Sender:Content-Transfer-Encoding:
	Content-Type:Cc:List-Subscribe:List-Help:List-Post:List-Archive:
	List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:To:From:
	Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:Resent-Sender
	:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:List-Owner;
	bh=EHf5gsh68g59KK8yNTSa38Wrpeqi0pX/1zz1O1VALQc=; b=ispz/miks6ImhpuIaBlgXhkl+x
	MyqM1FAVCnO8unjCcXVisMTjPGgkmFCV76lJTk85NFmTSCAWZ5TyzMPHqVeD8VJAgVn3Igbk9VlYw
	1EaXgpUB1B2YioTF01kxKjenF3lNc7qyRo33SevIH+3RFZU8vvsUJ3uIGToxqXzOP87gkiFRWS2iC
	TJ8wmhfwox9o0TJKwSK3X7gkLNwQQiigEwz46+CgstomEeaSz9ip9qZRuDOWsH51RdfJSrxaYK7jH
	ErsvZtPJa3xWrbs3TRVPMnwrMed9aJaTkxyver0aPmy7Mqt9SRMWSZ4on9M47Mbd3E3hud49+hF8x
	NYTSMhsA==;
Received: from localhost ([::1] helo=merlin.infradead.org)
	by merlin.infradead.org with esmtp (Exim 4.92.3 #3 (Red Hat Linux))
	id 1lBufc-0008SH-9p; Tue, 16 Feb 2021 07:21:16 +0000
Received: from mail-wm1-x330.google.com ([2a00:1450:4864:20::330])
 by merlin.infradead.org with esmtps (Exim 4.92.3 #3 (Red Hat Linux))
 id 1lBufZ-0008RX-6B
 for openwrt-devel@lists.openwrt.org; Tue, 16 Feb 2021 07:21:14 +0000
Received: by mail-wm1-x330.google.com with SMTP id j11so8214004wmi.3
 for <openwrt-devel@lists.openwrt.org>; Mon, 15 Feb 2021 23:21:11 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=Uzappv7BEgkVlVL4vmgYiVT3B/5lIacJZX3LpiYLuUU=;
 b=oUlg029Bv/0Dq+lHWXfWV889An/jRXiLM9Kr4ww3hItoClPbWPUTh4Ezi6XpwmNMud
 xGW7XQoAh/zCHNl5/KmmHqRxeJtW7z8oql0jqdeCK6+taxqPoVi2zFBlHOZShlrqQ4RY
 GwZy2puovmebgmlcJRJbppjGlkrxvUhnhtieFwUT2bSN2R7yEfy1iEB7VXllilwJsM4R
 ODix8BKWDPBCEQo4hCLHF49Q0cOkp5g00zVE28lCS5wMxzXgokpJcUFFqlFNrNy1jiox
 Ui3E+8qmPLL7IMFgFtXeQXn6KtRAg3+z6EUCLId/mSNEKDMUiG4xDNiGX+17NXdjoxfx
 ylOg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=Uzappv7BEgkVlVL4vmgYiVT3B/5lIacJZX3LpiYLuUU=;
 b=VDCWPZ7W33lqIt0SZzwMLswdF4qU3Zu/fVBzOx2JQTYgJUm/Jk1XWJnJeAt/HxeuFs
 EboHPUpsT2mHutVBSw+UUFjYvKbNpeueis/GN/rQAUAcvRvTkEyQxrgLdn5Xo5NUJJV6
 eFAx5kMVPu0b2+xvyz3is2bh7v4CowbPpZ4Hib+FasbsDfh+ZSRKj+1D90s14dsZgOmt
 kYBoemeQIEGdRcOmJBflUF1qWdcvTcd7QvpRl64KdrU249xz9LYHsb968rsElBeX29no
 Rx+Mhfur7Q934PgNFw6Ld/VH2zuFZ3Lmy1Lct68yWREceHkH1pKMJTcWcNFlCFLeLdvr
 HEgQ==
X-Gm-Message-State: AOAM5323BKLVqUjhBSErfIGwGiO/5I1CBcpZMDNFxj8djuWKHcsuWGMk
 2GS4hC8fVIwRRD58VzPS/qOFRX52Lj8=
X-Google-Smtp-Source: 
 ABdhPJyJ4IxXIEU9pPPLF00X0AyzJus8Pkxde21sKXLK4+msaij97zr1stzrtFeVoGeUEbC7EKTNJQ==
X-Received: by 2002:a7b:c308:: with SMTP id k8mr1959585wmj.54.1613460070242;
 Mon, 15 Feb 2021 23:21:10 -0800 (PST)
Received: from mamamia.internal (a89-182-179-14.net-htp.de. [89.182.179.14])
 by smtp.gmail.com with ESMTPSA id i10sm35614368wrp.0.2021.02.15.23.21.09
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Mon, 15 Feb 2021 23:21:09 -0800 (PST)
From: Andre Heider <a.heider@gmail.com>
To: openwrt-devel@lists.openwrt.org
Subject: [PATCH] build: prereq: drop support for Python 3.5
Date: Tue, 16 Feb 2021 08:21:08 +0100
Message-Id: <20210216072108.26107-1-a.heider@gmail.com>
X-Mailer: git-send-email 2.30.0
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20210216_022113_258965_641BD3D8 
X-CRM114-Status: GOOD (  10.98  )
X-Spam-Score: -0.2 (/)
X-Spam-Report: SpamAssassin version 3.4.4 on merlin.infradead.org summary:
 Content analysis details:   (-0.2 points)
 pts rule name              description
 ---- ----------------------
 --------------------------------------------------
 0.0 SPF_HELO_NONE          SPF: HELO does not publish an SPF Record
 -0.0 SPF_PASS               SPF: sender matches SPF record
 0.0 FREEMAIL_FROM          Sender email is commonly abused enduser mail
 provider [a.heider[at]gmail.com]
 -0.0 RCVD_IN_DNSWL_NONE     RBL: Sender listed at https://www.dnswl.org/,
 no trust [2a00:1450:4864:20:0:0:0:330 listed in]
 [list.dnswl.org]
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from
 author's domain
 0.1 DKIM_SIGNED            Message has a DKIM or DK signature,
 not necessarily
 valid
 -0.1 DKIM_VALID Message has at least one valid DKIM or DK signature
 -0.1 DKIM_VALID_EF          Message has a valid DKIM or DK signature from
 envelope-from domain
X-BeenThere: openwrt-devel@lists.openwrt.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: OpenWrt Development List <openwrt-devel.lists.openwrt.org>
List-Unsubscribe: <https://lists.openwrt.org/mailman/options/openwrt-devel>,
 <mailto:openwrt-devel-request@lists.openwrt.org?subject=unsubscribe>
List-Archive: <http://lists.openwrt.org/pipermail/openwrt-devel/>
List-Post: <mailto:openwrt-devel@lists.openwrt.org>
List-Help: <mailto:openwrt-devel-request@lists.openwrt.org?subject=help>
List-Subscribe: <https://lists.openwrt.org/mailman/listinfo/openwrt-devel>,
 <mailto:openwrt-devel-request@lists.openwrt.org?subject=subscribe>
Cc: Rosen Penev <rosenp@gmail.com>
Sender: "openwrt-devel" <openwrt-devel-bounces@lists.openwrt.org>
Errors-To: 
 openwrt-devel-bounces+incoming=patchwork.ozlabs.org@lists.openwrt.org

The final Python 3.5 release was 3.5.10 in September 2020 [0].
This release series is now End-of-Life (EOL).

The only LTS distribution that still only ships Python 3.5 is Ubuntu 16.04,
which will be EOL in April 2021 [1].

The meson build system bumped their python requirement to 3.6 for the 0.57.0
release. This patch ensures that OpenWrt can update meson while still
relying on the host python.

[0] https://www.python.org/dev/peps/pep-0478/#id4
[1] https://ubuntu.com/about/release-cycle
[2] https://mesonbuild.com/Release-notes-for-0-57-0.html#minimum-required-python-version-updated-to-36

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 include/prereq-build.mk | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/include/prereq-build.mk b/include/prereq-build.mk
index f88e853b52..88064abf98 100644
--- a/include/prereq-build.mk
+++ b/include/prereq-build.mk
@@ -141,21 +141,19 @@ $(eval $(call SetupHostCommand,perl,Please install Perl 5.x, \
 
 $(eval $(call CleanupPython2))
 
-$(eval $(call SetupHostCommand,python,Please install Python >= 3.5, \
+$(eval $(call SetupHostCommand,python,Please install Python >= 3.6, \
 	python3.9 -V 2>&1 | grep 'Python 3', \
 	python3.8 -V 2>&1 | grep 'Python 3', \
 	python3.7 -V 2>&1 | grep 'Python 3', \
 	python3.6 -V 2>&1 | grep 'Python 3', \
-	python3.5 -V 2>&1 | grep 'Python 3', \
-	python3 -V 2>&1 | grep -E 'Python 3\.[5-9]\.?'))
+	python3 -V 2>&1 | grep -E 'Python 3\.[6-9]\.?'))
 
-$(eval $(call SetupHostCommand,python3,Please install Python >= 3.5, \
+$(eval $(call SetupHostCommand,python3,Please install Python >= 3.6, \
 	python3.9 -V 2>&1 | grep 'Python 3', \
 	python3.8 -V 2>&1 | grep 'Python 3', \
 	python3.7 -V 2>&1 | grep 'Python 3', \
 	python3.6 -V 2>&1 | grep 'Python 3', \
-	python3.5 -V 2>&1 | grep 'Python 3', \
-	python3 -V 2>&1 | grep -E 'Python 3\.[5-9]\.?'))
+	python3 -V 2>&1 | grep -E 'Python 3\.[6-9]\.?'))
 
 $(eval $(call SetupHostCommand,git,Please install Git (git-core) >= 1.7.12.2, \
 	git --exec-path | xargs -I % -- grep -q -- --recursive %/git-submodule))
